google/datalab/pipeline/_pipeline.py

```diff
@@ -0,0 +1,211 @@
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+
+from google.datalab import utils
+from enum import Enum
+
+
+class Operator(Enum):
+  """ Represents a mapping from the Airflow operator class name suffix (i.e. the
+  portion of the class name before the "Operator", and the corresponding string
+  used in the yaml config (in the cell-body of '%pipeline create'). This
+  mapping enables us to onboard additional Airflow operators with minimal code
+  changes.
+  """
+  BigQuery = 'bq'
+  BigQueryTableDelete = 'bq-table-delete'
+  BigQueryToBigQuery = 'bq-to-bq'
+  BigQueryToCloudStorage = 'bq-to-gcs'
+  Bash = 'bash'
+
+
+class Pipeline(object):
```

Comment:
I view this class as a broad entry-point for all pipeline related activities in Datalab, with code generation being one of them. So I've named it after the magic itself. I feel *Generator would make it too narrow and susceptible to future change.